lumicks/pylake/scan.py

```diff
@@ -14,16 +15,19 @@ class Scan(Kymo):
     file : lumicks.pylake.File
         The parent file. Used to loop up channel data
     """
-    def __init__(self, h5py_dset, file):
-        super().__init__(h5py_dset, file)
-        self._num_frames = self.json["scan count"]
-        if len(self.json["scan volume"]["scan axes"]) > 2:
+    def __init__(self, name, file, start, stop, metadata):
+        super().__init__( name, file, start, stop, metadata)
+        self._num_frames = self.metadata["scan count"]
+        if len(self.metadata["scan volume"]["scan axes"]) > 2:
             raise RuntimeError("3D scans are not supported")
 
     def __repr__(self):
         name = self.__class__.__name__
         return f"{name}(pixels=({self.pixels_per_line}, {self.lines_per_frame}))"
 
+    def __getitem__(self, item):
+        raise NotImplementedError("Indexing and slicing are not implemented for scans")
```

Comment:
Yeah, I agree that scan does sound less specialized than Kymo does. But we'd also have to remove specializations in the other direction however (such as multiple frames), so I'm not sure whether a refactor would help much on that front. Mostly I think the lack of any test for Scan caused this regression.

I'm also a bit hesitant to refactor it until we have higher coverage in this area. I'd be for doing the bugfix first, then increasing test coverage and then perhaps refactoring it when we have better coverage for these two (and other) classes in the codebase.

Reply:
> We should also take this opportunity to take into account @dean0x7d's[ recent comment](https://github.com/lumicks/pylake/pull/16#discussion_r320623975) on one of the closed PRs.

I have removed the convenience function for setting limits as requested in the pull request comment.

Reply:
`Scan` is inheriting `Kymo` is very much a result of me rushing the initial implementation of `pylake`. I regret it. Neither should inherit from the other, it's too confusing. The best thing to do is to have two fully separate classes and deduplicate code by spinning out helper functions/classes as needed.

Reply:
I'm fine with refactoring it. Do we want to piggyback a refactor of Kymo and Scan on this PR? Or should I open a new issue for refactoring them?

Reply:
I would make it a separate PR. Same for the coverage tool.

Reply:
Alright. No prob! The added tests for scan could still be part of this one I suppose, since this regression brought up the need for more testing coverage there.